doc/contribute/coding_guidelines/index.rst

```diff
@@ -1297,6 +1297,85 @@ macros. Given that Zephyr uses a fork of the corresponding upstream for each
 module, it is always possible to patch the macro implementation in each module
 to avoid collisions.
 
+.. _coding_guideline_libc_usage_restrictions_in_zephyr_kernel:
```

Comment:
We should put something in here that any exception should require an implementation in `lib/libc/common` and test in `tests/lib/c_lib/` so as to provide an implementation that can be used for out of tree libc's that may not support the function.

Reply:
Added the following:

```
All non-ISO functions and macros listed above must be implemented by the
:ref:`common libc <c_library_common>` in order to make sure that these
functions can be made available when using a C standard library that does not
implement these functions.
```